bin/install

```diff
@@ -18,23 +19,109 @@ error_exit() {
   exit "${2:-1}"
 }
 
-platform() {
+get_platform() {
   local -r os=$(
     case "$(uname | tr '[:upper:]' '[:lower:]')" in
       darwin) echo "osx" ;;
       *) echo "linux" ;;
     esac
   )
 
-  echo "${os}-amd64"
+  echo "${os}"
+  return
 }
+get_arch(){
+  declare arch="$(uname -m)"
+  if [ "$arch" == 'x86_64' ]; then
+    echo '64'
+  elif [ "$arch" == 'i386' ]; then
+    echo '32'
+  elif [ "$arch" == 'i686' ]; then
+    echo '32'
+  else
+    error_exit 'There are no official releases for your architecture'
+  fi
+  return
+}
+
+get_assets_url() {
+  declare install_version="$1"
+
+  declare releases_json=$(curl -s "$RELEASES_URL")
+
+  declare -a asset_urls=($(echo $releases_json | grep -oP "(?<=\"assets_url\": \")[^\"]{8,}(?=\",)"))
+  declare -a tag_names=( $(echo $releases_json | grep -oP "(?<=\"tag_name\": \")[^\"]{3,}(?=\",)"))
+
+  for ((index=0; index<${#tag_names[@]}; index++)); do
+    if [ "${tag_names[$index]}" = "$install_version" ]; then
+      echo "${asset_urls[$index]}"
+      return
+    fi
+  done
+  error_exit 'Given version did not match any releases. Try list-all to see available options'
+}
+find_all_asset_names() {
+  declare install_version="$1"
+
+  declare assets_url="$(get_assets_url $install_version)"
+  if [ "$assets_url" == "" ]; then
+    error_exit "No URL found to release $install_version"
+  fi
+  declare assets_json="$(curl -s $assets_url)"
+  declare -a asset_names=($(echo $assets_json | grep -oP "(?<=\"name\": \")[^\"]{3,}(?=\",)"))
+  echo ${asset_names[@]}
+  return
+}
+filter_assets() {
```

Comment:
`filter_assets(inArr,platform,arch)` takes an array of release asset names, and returns all that match the given platform and architecture as an array.

The `platform` argument takes any value returned by `get_platform()`, and the `arch` argument takes any value returned by `get_arch()`. Other values have not been tested.

Values which do not match the following perl-style regexes are excluded:

- `$platform.*((86(?!_64))|32)` for an `arch` value of `32`
- `$platform.*(64)` for an `arch` value of `64`